### PR TITLE
Fix eks cluster PRC replace

### PR DESCRIPTION
**Pull request by VenelinMartinov**
The eks:Cluster resource has received a new parameter with ForceNew and a default. When upgrading from an old version before the parameter was added this triggers a replace.

This PR adds a workaround for that - when the state is read the `bootstrapSelfManagedAddons` parameter is added with its default value if not present.

covered by `TestEKSClusterUpgrade`

partially fixes https://github.com/pulumi/pulumi-aws/issues/4410

stacked on https://github.com/pulumi/pulumi-aws/pull/4403

### CODE REVIEW

File: provider/resources.go [hunk 1]
```diff
@@ -2192,6 +2192,16 @@ compatibility shim in favor of the new "name" field.`)
 					"node_group_name": tfbridge.AutoName("nodeGroupName", 255, "-"),
 				},
 			},
+			"aws_eks_cluster": {
+				TransformFromState: func(_ context.Context, pm resource.PropertyMap) (resource.PropertyMap, error) {
```
Comment (flostadler) [inline provider/resources.go:2196]: I might be missing something, but why is the State Upgrader not working here? Upstream configures it here: 
- https://github.com/hashicorp/terraform-provider-aws/blob/4b740e8f8919ff1b4571e15b286fc574456bd56d/internal/service/eks/cluster_migrate.go#L260
- https://github.com/hashicorp/terraform-provider-aws/blob/4b740e8f8919ff1b4571e15b286fc574456bd56d/internal/service/eks/cluster.go#L50

Comment (VenelinMartinov) [reply provider/resources.go:2196]: Ah, great find! This is likely https://github.com/pulumi/pulumi-terraform-bridge/pull/2081

Comment (VenelinMartinov) [reply provider/resources.go:2196]: That is still not rolled out to AWS. We could roll it out together with PRC.

Comment (flostadler) [reply provider/resources.go:2196]: Can we opt in specific resources only to https://github.com/pulumi/pulumi-terraform-bridge/pull/2081?

I think both are great enhancements, but rolling both out at the same time makes it harder to reason about potential issues that come up

Comment (VenelinMartinov) [reply provider/resources.go:2196]: No we can't filter which resources to enable https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 for.

I have confirmed that indeed that fixes the EKS replacement.

**Comment by VenelinMartinov** [comment]
closing in favour of https://github.com/pulumi/pulumi-aws/pull/4416, thanks @flostadler!

File: provider/resources.go (hunk 1 above)
Comment (flostadler) [reply provider/resources.go:2196]: I'd vote for rolling them out separately. https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 Seems relatively straight forward to roll out (famous last words), so wdyt about doing that right away?

Comment (VenelinMartinov) [reply provider/resources.go:2196]: I am unsure of the effects of https://github.com/pulumi/pulumi-terraform-bridge/pull/2081 without PRC - we have not tested that anywhere, so I'd be hesitant to do that.

Comment (VenelinMartinov) [reply provider/resources.go:2196]: There were concerns that 2081 might uncover latent panics which were hidden. The non-PRC bridge is known to transform some properties incorrectly, so running upgrades which were not run before could conceivably uncover some issues.

Comment (VenelinMartinov) [reply provider/resources.go:2196]: If rolling the two together is a big concern, we could fix the EKS cluster as in this PR.

That might cause similar issues in other resources though.

**Comment by VenelinMartinov** [comment]
4416 did not fix the issue: https://github.com/pulumi/pulumi-aws/issues/4410#issuecomment-2317651624

Reopening this.

**Review by t0yv0 — APPROVED** [review]
Per our discussion today, the root cause is https://github.com/pulumi/pulumi-terraform-bridge/issues/1667 that's not sending the right data into state migrations yes? Worth mentioning in the comment why this change is added, linking the root cause.

**Pull request merged**